cns/fsnotify/fsnotify.go

```diff
@@ -0,0 +1,141 @@
+package fsnotify
+
+import (
+	"context"
+	"io/fs"
+	"os"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type cnsclient interface {
+	ReleaseIPs(ctx context.Context, ipconfig cns.IPConfigsRequest) error
+}
+
+type Watcher struct {
+	CnsClient cnsclient
+}
+
+func WatchFs(w *Watcher, path, directory string, logger *zap.Logger) error {
+	// Create new fs watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Error("Error creating watcher: ", zap.Error(err))
+	}
+	defer watcher.Close()
+
+	// Start listening for events.
+	logger.Info("Listening for events from watcher")
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				logger.Info("Watcher: ", zap.Any("event: ", event.Name))
+				if event.Has(fsnotify.Create) {
+					logger.Info("created file: ", zap.Any("event: ", event.Name))
+					w.releaseIP(event.Name, path, directory, logger)
+				}
+				if event.Has(fsnotify.Remove) {
+					logger.Info("removed file: ", zap.Any("event: ", event.Name))
+				}
+			case watcherErr, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				logger.Error("Watcher Error: ", zap.Error(watcherErr))
+			}
+		}
+	}()
+
+	// Add directory where intended deletes are kept
+	dirpath := path + directory
+	err = os.MkdirAll(dirpath, 0o755) //nolint
+	if err != nil {
+		logger.Error("Error making directory: ", zap.Error(err))
+	}
+	err = watcher.Add(dirpath)
+	if err != nil {
+		logger.Error("Watcher add directory error: ", zap.Error(err))
+	}
+
+	// list the directory then call ReleaseIPs
+	logger.Info("Listing directory deleteIDs: ")
+	dirContents, err := os.ReadDir(dirpath)
+	if err != nil {
+		logger.Error("Error reading deleteID directory", zap.Error(err))
+	} else {
+		for _, file := range dirContents {
+			logger.Info("File to be deleted: ", zap.String("name", file.Name()))
+			logger.Info("Path to be removed from: ", zap.String("path: ", path))
+			w.releaseIP(file.Name(), path, directory, logger)
+		}
+	}
+
+	return nil
+}
+
+// WatcherAddFile creates new file using the containerID as name
+func WatcherAddFile(containerID, path, directory string) error {
```

Comment:
these methods can probably be `AddFile`/`RemoveFile` since the package is always part of the usage (ex `fsnotify.AddFile`)